2021/day10/day10.java

```diff
@@ -0,0 +1,83 @@
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Scanner;
+import java.util.Collections;
+import java.util.LinkedList;
+
+public class day10 {
+    public static ArrayList<ArrayList<Character>> readInput(String filename) {
+        ArrayList<ArrayList<Character>> input = new ArrayList<>();
+        Scanner sc = null;
+        try {
+            sc = new Scanner(new File(filename));
+            while (sc.hasNextLine()) {
+                ArrayList<Character> line = new ArrayList<>();
+                String s = sc.next();
+                for (int i = 0; i < s.length(); i++) {
+                    line.add(s.charAt(i));
+                }
+                input.add(line);
+            }
+        } catch (Exception e) {
+            System.out.println("Something went wrong" + e);
+        } finally {
+            sc.close();
+        }
+        return input;
+    }
+
+    public static void syntaxScoring() {
+        ArrayList<ArrayList<Character>> input = readInput("data.txt");
+        ArrayList<Long> autocompleteScores = new ArrayList<>();
+        int syntaxScore = 0;
+
+        for (ArrayList<Character> line : input) {
+
+            LinkedList<Character> stack = new LinkedList<>();
+            Boolean isCorrupted = false;
+            long autoTmpScore = 0;
+
+            for (int i = 0; i < line.size(); i++) {
+                if (line.get(i) == '(' || line.get(i) == '<' ||
```

Comment:
I would create two ArrayLists:
- `opening` chars
- `closing` chars
(my naming convention is awful, you can use something better if you want). Then I would just test if `opening.contains(c)`